### PR TITLE
MM-15850/MM-15851 GetPosts Before/After sync by default

### LINKED ISSUE #10977: Migrate "Post.GetPostsAfter" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-15851)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `GetPostsAfter` in the `Post` store.

The expected way to implement it is, go to the `Post` store implementation in the `store/sqlstore/` directory, modify the method `GetPostsAfter` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### LINKED ISSUE #10978: Migrate "Post.GetPostsBefore" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-15850)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `GetPostsBefore` in the `Post` store.

The expected way to implement it is, go to the `Post` store implementation in the `store/sqlstore/` directory, modify the method `GetPostsBefore` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### PR DESCRIPTION
**Pull request by tomasmik**
#### Summary
Both methods for getting posts before and after have been updatted.
Now they return a list of posts and an error.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-15851
https://mattermost.atlassian.net/browse/MM-15850

Fixes #10977
Fixes #10978

### CODE REVIEW

File: store/sqlstore/post_store.go [hunk 1]
```diff
@@ -583,30 +583,29 @@ func (s *SqlPostStore) GetPostsSince(channelId string, time int64, allowFromCach
...
 
-		var posts []*model.Post
-		var parents []*model.Post
-		_, err1 := s.GetReplica().Select(&posts,
-			`SELECT
+	var posts []*model.Post
+	var parents []*model.Post
+	_, err := s.GetReplica().Select(&posts,
+		`SELECT
 			    *
```
Comment (jespino) [inline store/sqlstore/post_store.go]: Nit: maybe you can align this SQL removing a tab from all the lines that are now shifted

File: store/sqlstore/post_store.go [hunk 2]
```diff
@@ -617,9 +616,12 @@ func (s *SqlPostStore) getPostsAround(channelId string, postId string, limit int
...
-			map[string]interface{}{"ChannelId": channelId, "PostId": postId, "Limit": limit, "Offset": offset})
-		_, err2 := s.GetReplica().Select(&parents,
-			`SELECT
+		map[string]interface{}{"ChannelId": channelId, "PostId": postId, "Limit": limit, "Offset": offset})
+	if err != nil {
+		return nil, model.NewAppError("SqlPostStore.GetPostContext", "store.sql_post.get_posts_around.get.app_error", nil, "channelId="+channelId+err.Error(), http.StatusInternalServerError)
+	}
+	_, err = s.GetReplica().Select(&parents,
+		`SELECT
 			    q2.*
```
Comment (jespino) [inline store/sqlstore/post_store.go]: Same here

**Comment by hanzei** [comment]
@tomasmik Would you please merge `master` into your branch?

**Comment by tomasmik** [comment]
I will merge master and fix indentation tomorrow. Just didn't have the time to pull this up.
Sorry

**Comment by hanzei** [comment]
No problem. I just wanted to make sure the next steps are clear.

**Comment by tomasmik** [comment]
Merged master, removed a single TAB from sql queries.

File: app/.#post.go [hunk 3]
```diff
@@ -0,0 +1 @@
+tomas@tomas.6171:1559840605
```
Comment (hanzei) [inline app/.#post.go]: This file should not be here :wink:

Comment (tomasmik) [reply app/.#post.go]: sorry that my IDE playing tricks. removed.

**Pull request merged**